cinema/tests/test_movie_api.py

```diff
@@ -157,3 +156,169 @@ def test_image_url_is_shown_on_movie_session_detail(self):
         res = self.client.get(MOVIE_SESSION_URL)
 
         self.assertIn("movie_image", res.data[0].keys())
+
+
+class UnauthenticatedMovieApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        response = self.client.get(MOVIE_URL)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class AuthenticatedMovieApiTest(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            "test@user.com", "testpassword"
+        )
+        self.client.force_authenticate(self.user)
+
+    def test_list_movie(self):
+        sample_movie()
+        movie_with_genres = sample_movie(title="Test Genres")
+        movie_with_actors = sample_movie(title="Test actors")
+
+        genre1 = sample_genre()
+        genre2 = sample_genre(name="Horror")
+
+        actor1 = sample_actor()
+        actor2 = sample_actor(first_name="Max")
+
+        movie_with_genres.genres.add(genre1, genre2)
+        movie_with_actors.actors.add(actor1, actor2)
+
+        response = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_filter_movies_by_genres(self):
+        movie1 = sample_movie(title="movie1")
+        movie2 = sample_movie(title="movie2")
+        movie3 = sample_movie(title="movie3")
+
+        genre1 = sample_genre()
+        genre2 = sample_genre(name="Horror")
+
+        movie1.genres.add(genre1)
+        movie2.genres.add(genre2)
+
+        response = self.client.get(MOVIE_URL, {"genres": f"{genre1.id}, {genre2.id}"})
+
+        serializer1 = MovieListSerializer(movie1)
+        serializer2 = MovieListSerializer(movie2)
+        serializer3 = MovieListSerializer(movie3)
+
+        self.assertIn(serializer1.data, response.data)
+        self.assertIn(serializer2.data, response.data)
+        self.assertNotIn(serializer3.data, response.data)
+
+    def test_filter_movies_by_actors(self):
+        movie1 = sample_movie(title="movie1")
+        movie2 = sample_movie(title="movie2")
+        movie3 = sample_movie(title="movie3")
+
+        actor1 = sample_actor()
+        actor2 = sample_actor(first_name="Actor2")
+
+        movie1.actors.add(actor1)
+        movie2.actors.add(actor2)
+
```

Comment:
as for me, will be better to use more descriptive names instead of movie1 and response
For example, movie_with_actor and filtered_movies. Smth like that